specification/eventhub/resource-manager/Microsoft.EventHub/preview/2018-01-01-preview/EventHub-preview.json

```diff
@@ -1,1411 +1,1679 @@
+          },
+          {
+            "$ref": "#/parameters/NamespaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/IPFilterRuleNameParameter"
+          },
+          {
+            "name": "parameters",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/IpFilterRule"
+            },
+            "description": "The Namespace IpFilterRule."
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Namespace IpFilterRule created",
+            "schema": {
+              "$ref": "#/definitions/IpFilterRule"
+            }
+          },
+          "default": {
+            "description": "Eventhub error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "delete": {
+        "tags": [
+          "Namespaces"
+        ],
+        "operationId": "Namespaces_DeleteIpFilterRule",
+        "x-ms-examples": {
+          "NameSpaceIpFilterRuleDelete": {
+            "$ref": "./examples/NameSpaces/IPFilterRule/EHNameSpaceIpFilterRuleDelete.json"
+          }
+        },
+        "description": "Deletes an IpFilterRule for a Namespace.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/NamespaceNameParameter"
+          },
+          {
+            "$ref": "#/parameters/IPFilterRuleNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          }
+        ],
+        "responses": {
+          "204": {
+            "description": "No content."
+          },
+          "200": {
+            "description": "Namespace IpFilterRule successfully deleted."
+          },
+          "default": {
+            "description": "Eventhub error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        }
+      },
+      "get": {
+        "tags": [
+          "Namespaces"
+        ],
+        "operationId": "Namespaces_GetIpFilterRule",
+        "x-ms-examples": {
+          "NameSpaceIpFilterRuleGet": {
+            "$ref": "./examples/NameSpaces/IPFilterRule/EHNameSpaceIpFilterRuleGet.json"
+          }
+        },
+        "description": "Gets an IpFilterRule for a Namespace by rule name.",
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/NamespaceNameParamet
```

Comment:
The resource type name should be plural: networkRuleset**s**. I'm not sure what the distinction is between virtualNetworkRules and networkRulsets, but could an improved resource type name help make that distinction?

Reply:
as this is the preview version, we will be will be deprecated virtualNetworkRules API and will be going ahead with only networkruleset API. 
for  plural: networkRulesets,  Correct me if I am wrong.  each namespace will be having only one networkRuleset which will consist of ipRule list and virtualNetworkRule list. 
so should we make plural: networkRulesets or singular : networkRuleset ?

Reply:
yeah, today we do pluralize the resourceType names even though for that type there may be just a single instance possible.